server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2818,6 +2822,26 @@ public UserVm destroyVm(DestroyVMCmd cmd) throws ResourceUnavailableException, C
         return destroyedVm;
     }
 
+    protected void removeTagsFromVm(long vmId) {
+        UserVmVO vm = _vmDao.findById(vmId);
+        Long resourceId = _taggedResourceService.getResourceId(vm.getUuid(), ResourceTag.ResourceObjectType.UserVm);
+        List<? extends ResourceTag> resourceTags = _taggedResourceService.listByResourceTypeAndId(ResourceTag.ResourceObjectType.UserVm, resourceId);
+        if (resourceTags.size() > 0) {
+            List<String> resourceIds = Arrays.asList(vm.getUuid());
+            Map<String, String> tags = new HashMap<>();
+            _taggedResourceService.deleteTags(resourceIds, ResourceTag.ResourceObjectType.UserVm, tags);
+            s_logger.debug("Tags from vm id: " + vmId + " has been removed.");
```

Comment:
what about adding the deleted tags to the debug message as well?

Reply:
Also, are you sending an empty list of tags for the deleteTags method?

Reply:
Yes, if someone would like debug the method and i need send a void map to method work.

Reply:
That last parameter can be null

Reply:
Can you add the deleted tags to the message?

Reply:
My bad, i'll change.

Reply:
i fixed.